macros/src/derives/format/codegen/fields.rs

```diff
@@ -37,21 +38,30 @@ pub(crate) fn codegen(
         }
 
         let format_opt = get_defmt_format_option(field)?;
-        let ty = as_native_type(&field.ty).unwrap_or_else(|| consts::TYPE_FORMAT.to_string());
+        let ty = as_native_type(&field.ty);
+        let field_ty = if ty.is_none() { Some(&field.ty) } else { None };
+        let ty = ty.unwrap_or_else(|| consts::TYPE_FORMAT.to_string());
```

Comment:
Could you please add a comment explaining what is happening in this code block?

Reply:
Done